server/routes/readingHistory.js

```diff
@@ -25,7 +25,7 @@ router.use(async (req, res) => {
 
     const docMeta = getMeta(res.locals.docId)
     const userInfo = req.userInfo
-    if (!docMeta || !userInfo) return
+    if (!docMeta || !userInfo || !datastoreClient) return
```

Comment:
Is this check being added for the first time we set datastoreClient on line 20?

Most of the time line 17 should keep us from reaching here, but if `getDatastoreClient()` returns null we'd still continue to this point. Should we consider aborting `recordView` right below line 20 in that case?